### PR TITLE
add hypercore support

### LINKED ISSUE #2: support hypercores
It'd be cool if this also supported `hypercore` as we could then plug it into our `hyperops` tools

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Add support for hypercores, closes GH-2

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,16 +4,16 @@ const discovery = require('hyperdiscovery')
 
 module.exports = function (archiveOrKey, opts) {
   var archive = archiveOrKey.key ? archiveOrKey : createArchive(archiveOrKey)
+  var feed = (archive.content) ? archive.content : archive
   var swarm = discovery(archive, opts)
+
   archive.open(function () {
```
Comment (joehand) [inline index.js:10]: Should this be feed.open? Not sure that'd work on archives if content isn't defined yet though.

File: index.js [hunk 2]
```diff
@@ -4,16 +4,16 @@ const discovery = require('hyperdiscovery')
 
 module.exports = function (archiveOrKey, opts) {
   var archive = archiveOrKey.key ? archiveOrKey : createArchive(archiveOrKey)
+  var feed = (archive.content) ? archive.content : archive
```
Comment (joehand) [inline index.js:7]: `archive.content` may be undefined here even though its an archive (if the `createArchive` function was used).

File: index.js [hunk 3]
```diff
@@ -4,16 +4,16 @@ const discovery = require('hyperdiscovery')
 
 module.exports = function (archiveOrKey, opts) {
   var archive = archiveOrKey.key ? archiveOrKey : createArchive(archiveOrKey)
```
Comment (joehand) [inline index.js:6]: It'd be nice if you could pass in a hypercore key too, instead of just the feed.

**Review by joehand — CHANGES_REQUESTED** [review]
This isn't working with the CLI library anymore running `hyperhealth <key>` throws an error:

```
    for (var i = 0; i < feed.peers.length; i++) {
                                  ^

TypeError: Cannot read property 'length' of undefined
    at Object.get (/Users/joe/node_modules/hyperhealth/index.js:20:35)
```

File: index.js (hunk 3 above)
Comment (yoshuawuyts) [reply index.js:6]: not sure how to detect if something is a hypercore or not based on a key - suggestions?

Comment (yoshuawuyts) [reply index.js:6]: believe we talked about this on IRC but not sure if we had a conclusive answer

File: index.js (hunk 2 above)
Comment (yoshuawuyts) [reply index.js:7]: iirc `hyperhealth` does nothing in that case - e.g. `archive.content` is needed for https://github.com/karissa/hyperhealth/blob/master/index.js#L36-L37 - what should the value of `.content` be?

Comment (joehand) [reply index.js:7]: It doesn't do anything while [content isn't populated](https://github.com/karissa/hyperhealth/blob/master/index.js#L16). But once content [gets populated after the open](https://github.com/karissa/hyperhealth/blob/master/index.js#L9) (when a connection succeeds and metadata is fetched) then it starts to return the health.

File: index.js (hunk 3 above)
Comment (joehand) [reply index.js:6]: You can use something [like this](https://github.com/mafintosh/hypercore-archiver/blob/master/index.js#L206). But there may be a more straightforward implementation I can try but need to go get ☕️ first :).

Comment (yoshuawuyts) [reply index.js:6]: as discussed on IRC a generic package might be neat to have to deal with this :D

File: index.js (hunk 1 above)
Comment (yoshuawuyts) [reply index.js:10]: not sure, my knowledge of `hyperdrive` is limited - the logic is the same as before tho so I'd rather not deal with changing it in this patch hmm

**Comment by yoshuawuyts** [comment]
updated!

**Comment by joehand** [comment]
Done in #5. Made node API a bit easier in that PR =).

**Pull request closed without merging**